### PR TITLE
Move check for upgraded client state height to 02-client layer.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

closes: #4212, ~~removing this check in the 08-wasm module is additionally required.~~ (Removed check in 08-wasm module)

### Commit Message / Changelog Entry

```bash
chore: move check for upgraded client state height in 02-client layer.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/02-client/keeper/client_test.go
Comment (DimitrisJim) [inline modules/core/02-client/keeper/client_test.go:452]: note that I added a test case here but additionally left the others present in 07-tendermint. (which also need tweaking since they fall victim to the whole not checking error returned and thereby not always checking what they are supposed to)

**Comment by crodriguezvega** [comment]
Thanks for picking this up, @DimitrisJim. Since the check needs to be removed in both 07-tendermint and 08-wasm, I think it would be best to wait until we merge 08-wasm to main and then we can fully address the whole issue in a single PR. What do you think?

**Comment by DimitrisJim** [comment]
> I think it would be best to wait until we merge 08-wasm to main and then we can fully address the whole issue in a single PR. What do you think?

sounds perfectly fine to me, we can leave as is and I'll just rebase when the time is right

**Review by crodriguezvega — APPROVED** [review]
Looks fine to me. Happy that we can finally merge this one! :)

**Review by chatton — APPROVED** [review]
LGTM thanks @DimitrisJim

**Review by colin-axner — APPROVED** [review]
LGTM, the 02-client tests should really be refactored some day, seems like some rearrangement could remove a lot of code cov

**Pull request merged**
